pkg/machine/qemu/machine_test.go

```diff
@@ -36,5 +93,8 @@ func TestPropagateHostEnv(t *testing.T) {
 	tokens := strings.Split(cmdLine[1], ",string=")
 	decodeString, err := base64.StdEncoding.DecodeString(tokens[1])
 	assert.NoError(t, err)
-	assert.Equal(t, fmt.Sprintf("HTTP_PROXY=\"proxy\"|SSL_CERT_FILE=\"%s/foo.cert\"|SSL_CERT_DIR=%q", machine.UserCertsTargetPath, machine.UserCertsTargetPath), string(decodeString))
+
+	for _, item := range tests {
+		assert.Contains(t, string(decodeString), item.expect)
```

Comment:
Suggestion:
```go
    for _, item := range tests {
        switch item.expect {
        case "":    // expect it to be unset/empty/missing/FIXME
            FIXME: assert that decodeString does not contain item.key? Or that it contains "item.key=$" ?
        case "=":    // expect it to be exactly what we set it ti
            assert.Contains(t, string(decodeString), fmt.Sprintf("%s=%s", item.key, item.value)
        default:    // expect a transmogrification of what we set it to
            assert.Contains(t, string(decodeString), fmt.Sprintf("%s=%s", item.key, item.expect)
        }
    }
```
And, I don't know what all those '|' pipes are in the original code, if they're a delimiter, maybe it'd be better so `Split()` into an array, then run an array-contains-exact-string assertion instead of the vague `assert.Contains`.

But I really don't know this code, so grain of salt.

Reply:
Your advice was essential, thanks. I have optimized this code section, and you can now perform a new code review.